packages/blockchain/src/blockchain.ts

```diff
@@ -352,7 +355,7 @@ export class Blockchain implements BlockchainInterface {
 
       this.DEBUG &&
         this._debug(
-          `Canonical head set from ${bytesToHex(canonicalHeadHash)} to ${bytesToHex(hash!)} (number ${bigIntToHex(canonicalHead)})`,
+          `Canonical head set from ${canonicalHeadHash ? bytesToHex(canonicalHeadHash) : 'unknown'} to ${bytesToHex(hash!)} (number ${bigIntToHex(canonicalHead)})`,
```

Comment:
Since `this.DEBUG` is true in this case, we know `canonicalHeadHash` is set, so the `? :` logic should be deleted here (if TypeScript complains, we know `canonicalHeadHash` is set so you can make TS know it is set by using `canonicalHeadHash!` (add exclamation mark))

Reply:
Fixed